### PR TITLE
[ts-bodies] Destructuring, typeof, fixes

### PR DESCRIPTION
**Pull request by trilis**

### CODE REVIEW

File: compiler/test/data/typescriptBodies/ternaryOperator.kt [hunk 1]
```diff
@@ -18,7 +18,7 @@ import org.w3c.xhr.*
 fun f() {
     var i: Any = 1
     while (i <= 100) {
-        console.log(if (i % 15 == 0) "FizzBuzz" else if (i % 3 == 0) "Fizz" else if (i % 5 == 0) "Buzz" else "")
+        console.log(if ((i % 15 == 0)) "FizzBuzz" else (if ((i % 3 == 0)) "Fizz" else (if ((i % 5 == 0)) "Buzz" else "")))
```
Comment (shabunc) [inline compiler/test/data/typescriptBodies/ternaryOperator.kt:21]: Let's keep single brackets in this case

File: typescript/ts-node-lowering/src/org/jetrbains/dukat/nodeLowering/lowerings/convertStatements.kt [hunk 2]
```diff
@@ -502,6 +577,46 @@ private class ExpressionConverter() {
...
+                listOf(),
+                null,
+                null
+            )
+        )
+    }
+
+    private fun BinaryExpressionDeclaration.convert(): ExpressionModel {
+        if (convertBinaryOperator(operator) == EQ) {
+            val left = left
```
Comment (shabunc) [inline typescript/ts-node-lowering/src/org/jetrbains/dukat/nodeLowering/lowerings/convertStatements.kt:603]: why we need this variables?

Comment (trilis) [reply typescript/ts-node-lowering/src/org/jetrbains/dukat/nodeLowering/lowerings/convertStatements.kt:603]: to enable smart-casting in type checks below - is there more elegant way to do this?

File: compiler/test/data/typescriptBodies/ternaryOperator.kt (hunk 1 above)
Comment (trilis) [reply compiler/test/data/typescriptBodies/ternaryOperator.kt]: Fixed

**Comment by trilis** [comment]
ed0832116722f95cd429880e3d1e4ec46f2ad569

**Pull request closed without merging**